### DISCUSSION TITLE
Sonoff TRVZB 1.1.4 Firmware issues

### DISCUSSION THREAD

**Discussion by jamesonuk** [Koenkk/zigbee2mqtt#21694] [Devices]
The 1.1.4 firmware for the Sonoff TRVZB was published and then pulled at the request of Sonoff.

This is partly created to separate out comments on https://github.com/Koenkk/zigbee2mqtt/issues/19269 that are related to the 1.1.4 firmware rather than them getting lost in that issue or general discussion of the device.

Also check https://forum.ewelink.cc/t/trvzb-firmware-update-1-1-4-big-problem/29029 for info

Seems a lot of people have suggested a reset after upgrading sorts the device out but a few have said this hasn't worked.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1040621/ad5dcfcf-5c92-45e1-8ead-f07d03c5771b)

Please keep this related to the 1.1.4 firmware and whether it has caused / fixed issues.

**Comment by SentenciaLaMalice** [comment]
I think I missed some activity on this issue :( ... so, how can I install 1.1.4 with zigbee2mqtt?. I can't see any update available. 

Thanks in advance

**Reply by Korkman** [reply]
> and then pulled at the request of Sonoff

@jamesonuk Because of stability / quality issues or other reasons?

**Comment by SentenciaLaMalice** [comment]
It seems like 1.1.4 is again available for updating on z2m. Let's try ...

**Reply by jamesonuk** [reply]
The firmware is 1.1.5 so it different but you won't see the valve values until https://github.com/Koenkk/zigbee-herdsman-converters/pull/7130 gets merged.    Trying this on a less critical TRV now........

**Reply by GTunney** [reply]
Report back and let us know how you get on please? :D

**Reply by srett** [reply]
Ah weird, just noticed the notification in HA and wondered what version this would be, since I almost installed the troublesome 1.1.4 back in March.

For some reason z2m only displays the current version in the OTA table, and not the newly found one :facepalm:. And in the HA notification, it seems to use some other representation, it says current version 4353, latest version 4357.

There isn't an official changelog by any chance?

**Comment by jamesonuk** [comment]
> Report back and let us know how you get on please? :D

Firmware upgrade itself seemed to go OK and my very limited testing makes me think the TRV is functioning OK but I need to do some testing before I roll it out any further.

Adding the new valve attributes however gives an error

> 2024-04-08 17:29:36Publish 'get' 'valve opening' to 'Study TRV' failed: 'Error: Cluster with name '64529' does not exist'

Which I think is actually something with 1.36.1 rather than the change

**Reply by dktrekkie** [reply]
What about the calibration? Can you tell if that works? :)

**Reply by jamesonuk** [reply]
> What about the calibration? Can you tell if that works? :)

Needs more testing but has changed twice and seems OK so far

**Comment by rogerjak** [comment]
> Firmware upgrade itself seemed to go OK

I'm stuck on 98.7%, upgrading from 1.1.4 :(

Edit: The last 1.3% took 15 minutes, almost the same as the rest of it, but I made it :)

**Reply by jamesonuk** [reply]
> I'm stuck on 98.7%, upgrading from 1.1.4 :(

Yeah the last 1% seemed to take forever and got me.worried too but did get thete

**Comment by photomoose** [comment]
I've successfully installed 1.1.4 onto a single device and the local temperature calibration appears to be fixed. When changing the slider in the Z2M UI, I notice a longer delay before the local temperature value is updated, whereas with devices running the previous version, the local temperature is updated nearly immediately and goes berserk when the slider moves.

I'm also seeing the error about the cluster with name '64529' - this is actually cluster 0xFC11 and is used for the properties I exposed a few months back (child lock, open window, frost protection etc.). As @jamesonuk suggests, this problem is probably a Z2M issue and not a firmware issue as the problem exists on all my other devices running the older version of the firmware. This is likely to be a hex to string problem or something - I'll do some digging, unless @Koenkk happens to know of any recent change in this area?

As others have suggested, the upgrade does hang around the 98-99% position for a while - be patient and do not reset your TRV during this time!

**Reply by SentenciaLaMalice** [reply]
Just updated 9 devices using the Home Assistant functions for update (not the OTA update from Z2M) ... every single device takes about an hour to update!!!. My guess is that the update process takes place only on lapses when the device wakes up for reporting.

All devices are aparently working fine. Moving local calibration setting no longer derive on device going crazy!.

**Reply by elmystico** [reply]
This is 1.1.5 version apparently.
Updated one device seems ok but troubles are as you've mentioned z2mqtt seems in need of update

**Comment by GTunney** [comment]
I've upgraded to 1.1.5 and all seems ok, one thing I have noticed is that it seems to take longer to close / open the valve, It used to take about 2 seconds on mine and now the TRVs make a noise for about 5-7 seconds.

**Reply by jj-uk** [reply]
I bought these trvs specifically as they were fast. If they're now slow I'll skip the update.

Can anyone else confirm the reduced speed?

**Reply by photomoose** [reply]
Just tried several around the house - for me, they all take about 3 seconds to open and 3 seconds to close.

@GTunney I wonder whether it's worth you removing and re-inserting the battery and entering, what the manual says, "debug" mode - where I think the TRV works out the distance to travel from the closed to open positions. See if that makes any difference, perhaps?

**Reply by GTunney** [reply]
I have tried factory reset and also the debugging mode and neither have made any difference. I can hear it engage the TRV pin after about 2 seconds but the motor inside keeps going for a further 4 seconds after engaging the pin for some reason. Never happened on 1.1.4. Pretty much as soon as it engaged the pin it would stop.

**Comment by blacknell** [comment]
Does anyone have any further observations with the latest firmware? I'm nervous to update

**Reply by jamesonuk** [reply]
> Does anyone have any further observations with the latest firmware? I'm nervous to update
I upgraded nine TRVs two weeks ago and after that switched Better Thermostat from temperature based to offset based and they all seem to have been fine since.   

There are issues with the custom cluster and the latest version of Z2M which clouds the water a little (eg. you can't set the open window detection) but that is not around the device or firmware.

**Reply by photomoose** [reply]
Ah yes, I was going to look into that issue - I've got some free time today so I'll go investigate. But other than that, the latest firmware is working fine for me on all my TRVs.

**Comment by fir3drag0n** [comment]
Opening / closing degree ist now merged to dev branch:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/36398628/06ebce54-5bdf-4d60-b7d9-cbc1ff2cc406)

Did you already test the opening/closing degree?

**Comment by photomoose** [comment]
I've noticed that some refactoring work in the current (unreleased) master branch is now causing issues with being able to turn on child lock, open window detection etc. I've raised an [issue](https://github.com/Koenkk/zigbee-herdsman-converters/issues/7436) to track this.

**Comment by dktrekkie** [comment]
I have 3 TRV's that are still on 1.0.8. I can’t update them. They all just stop after a very long time. And now I only get:

Error 2024-04-24 16:37:13Update of 'Dining Room Thermostat' failed (OTA: Device didn't respond to OTA request)

Whrn I try again. I have factory reset also, but does not make a difference.

There is even one TRV that now shows as having no update. (Still on 1.0.8)?

I also tried to force update by adding:
```
  {
    "modelId": "TRVZB",
    "url": "trvzb_v1.1.5. ota",
    "path": "./trvzb_v1.1.5. ota",
    "force": true
  }
```
To index.json, and then refiring to that under setup. Gave me the same error.

How do I update them? Preferably directly to 1.1.5?

All the others were on 1.1.1, and they updated nicely to 1.1.5.

Anyone?

**Comment by srett** [comment]
I finally dared to hit the update button. First time around the update failed, it went to 100% but then said "invalid image".
Second attempt was successful, but for some reason the UI is still reporting
> Firmware build date    20230811
Firmware version    1.1.1

Same on the OTA page, but when I click "check for updates" it says "no update available". And changing the temperature offset doesn't make the local temperature go apeshit anymore. Also when power-cycling the TRV it now briefly displays "64" which I *think* it didn't do before... So I think it did update but for some reason Z2M doesn't show the proper version information, even after restart.

**Reply by fir3drag0n** [reply]
Same problem. Did you solve it?

**Reply by srett** [reply]
Not yet. I haven't updated z2m since then (I frequently tested the dev branch a while ago), but I would assume this is just some sort of display bug, since I cannot hit update again. Plus, changing the new options, like opening position, doesn't lead to any errors and seems to persist.

**Comment by podkilla** [comment]
So is it safe to update 1.1.4 -> 1.1.5 ? I skipped the update until now because of the 1.1.4 update problems.

**Reply by ChavezD** [reply]
After the update to 1.1.4 I had to reset, but the update to 1.1.5 went well immediately.

**Reply by janrothkegel** [reply]
@ChavezD How did you upgrade to 1.1.5?

**Reply by blacknell** [reply]
If you find the device in settings mqtt, there should be an update button waiting for you. Unless you disabled upgrade checking.

**Reply by ear9mrn** [reply]
For me, all my TRVZB are showing firmware 1.2.1 (build 20240823). I am struggling to get the valve opening degree to work. When I change the slider, how do i ensure/check that the parameter has been set?  As far as I can tell when I set the close degree to 0% then set the mode to off, the value appears to be fully closed, rather than fully open as that attribute suggest? Any help/advice would be appreciated. Thanks.

**Comment by ansgarkewitz** [comment]
Hello I am using the ZHA integration but the firmware version is displayed as unknown. Is there any way how to upgrade the firmware with ZHS

**Reply by dafunkydan** [reply]
@ansgarkewitz You are terribly wrong here. This is about Zigbee2mqtt.
